### PR TITLE
#5965: Create google PKCE integration definition

### LINKED ISSUE #5965: Google sheets slice 1.1: Define a OAuth2 PKCE Google Integration Definition
https://www.notion.so/pixiebrix/Make-Google-and-Google-Sheets-a-Standard-Integration-efb28821a2b54365a85bcd6981e9c71c?pvs=4#a11ed851029d4afbaee2de1df20d11bc

### PR DESCRIPTION
**Pull request by johnnymetz**
## What does this PR do?

- Closes #5965 

## Reviewer Tips

- I created two new OAuth2 client in the Google Cloud Console. The prod credentials are currently listed as the default in the YAML definition.

Screenshot 2023-06-26 at 5 08 22 PM

## Demo

https://www.loom.com/share/3385c712589343f8b12e179c44cb6398

## Team Coordination

- [x] After PR is merged, paste definition into Workshop and ensure it gets added as a `Package` obj in the database

## Checklist

- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: contrib/services/google-oauth2-pkce.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    clientId:
+      type: string
+      description: The OAuth 2.0 client ID
+      default: 676864882630-n33qt0a585mcaf2vithipddia0l8blmc.apps.googleusercontent.com
+    clientSecret:
+      type: string
+      description: The OAuth 2.0 client secret. Note that this "secret" can be shared publicly.
+      default: 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
+authentication:
+  # No baseURL is okay here, right?
```
Comment (johnnymetz) [inline contrib/services/google-oauth2-pkce.yaml:21]: Is no `baseURL` correct here? Note we're using this access token to hit multiple different google drive API's:

- `https://www.googleapis.com/drive/v3/files`: to get all google sheets user has access to
- `https://sheets.googleapis.com/v4/spreadsheets/<spreadsheet_id>`: to read /write to a single sheet

File: contrib/services/google-oauth2-pkce.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  oauth2:
+    # access_type=offline and prompt=consent will force a refresh token to be returned
```
Comment (johnnymetz) [inline contrib/services/google-oauth2-pkce.yaml:23]: Can I include comments in these yaml definitions? I didn't see a comment in any of our existing definitions so I'm guessing no. If not, I'll remove this comment.

File: contrib/services/google-oauth2-pkce.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  name: Google (Client Integration)
+  description: OAuth2 PKCE Authentication for Google
+inputSchema:
+  $schema: https://json-schema.org/draft/2019-09/schema#
+  type: object
+  properties:
+    clientId:
+      type: string
+      description: The OAuth 2.0 client ID
+      default: 676864882630-n33qt0a585mcaf2vithipddia0l8blmc.apps.googleusercontent.com
```
Comment (johnnymetz) [inline contrib/services/google-oauth2-pkce.yaml:15]: I'm using prod credentials as the default here. Another option is to use the dev credentials as the default.

**Comment by twschiller** [comment]
@johnnymetz the team coordination is not correct - this is a definition, so wouldn't show up in headers. Also we would either: 1/ put this in the app projects and load via load_configs, or 2/ define in the workshop

File: contrib/services/google-oauth2-pkce.yaml (hunk 3 above)
Comment (twschiller) [reply contrib/services/google-oauth2-pkce.yaml:15]: Prod should be the default. We don't want end-users to have to change anything. They won't know the values

File: contrib/services/google-oauth2-pkce.yaml (hunk 1 above)
Comment (twschiller) [reply contrib/services/google-oauth2-pkce.yaml:21]: Correct, we should leave off the base URL. Alternatively, you could pick the more common one

File: contrib/services/google-oauth2-pkce.yaml (hunk 2 above)
Comment (twschiller) [reply contrib/services/google-oauth2-pkce.yaml:23]: Comments are fine to include in YAML files

File: contrib/services/google-oauth2-pkce.yaml (hunk 3 above)
Comment (johnnymetz) [reply contrib/services/google-oauth2-pkce.yaml:15]: Ok. Also see my edit to the original comment. Do we even want to parameterize these values?

File: contrib/services/google-oauth2-pkce.yaml (hunk 1 above)
Comment (johnnymetz) [reply contrib/services/google-oauth2-pkce.yaml:21]: I'm going to leave off and include a comment. Putting one there may be confusing.

**Comment by johnnymetz** [comment]
> After we merge this, we'll want to add the definition in the workshop

Got it, I updated the "Team Coordination" section

File: contrib/services/google-oauth2-pkce.yaml (hunk 3 above)
Comment (twschiller) [reply contrib/services/google-oauth2-pkce.yaml:15]: > Do we even want to parameterize these values?

I don't think there's any reason to parameterize

**Comment by johnnymetz** [comment]
Ready for a final review

**Review by grahamlangford — APPROVED** [review]

**Review by grahamlangford — APPROVED** [review]

**Pull request merged**